packages/ui/src/components/va-form/VaForm.stories.ts

```diff
@@ -1,12 +1,443 @@
+        label="with error highlight"
+        :rules="[value => 'error']"
+      />
+    </va-form>
+  `,
+})
+
+HideErrors.play = async ({ step }) => {
+  const [firstInput, secondInput] = document.querySelectorAll('div.va-input-wrapper')
+
+  await step('Does not highlight invalid input', async () => {
+    expect(firstInput).not.toHaveClass('va-input-wrapper--error')
+  })
+
+  await step('Highlights invalid input', async () => {
+    expect(secondInput).toHaveClass('va-input-wrapper--error')
+  })
+}
+
+export const Focus = () => ({
+  components: { VaForm, VaInput, VaButton },
+  template: `
+    <va-form 
+      class="flex space-x-2 mb-6 max-w-sm" 
+      ref="focusForm"
+    >
+      <va-input
+        label="should be focused"
+      />
+      <va-input
+        label="should not be focused"
+      />
+    </va-form>
+    <va-button @click="$refs.focusForm.focus()">
+      Focus first input
+    </va-button>
+  `,
+})
+
+Focus.play = async ({ canvasElement, step }) => {
+  const canvas = within(canvasElement)
+  const firstInput = canvas.getByRole('textbox', { name: 'should be focused' }) as HTMLElement
+  const secondInput = canvas.getByRole('textbox', { name: 'should not be focused' }) as HTMLElement
+  const button = canvas.getByRole('button', { name: 'Focus first input' }) as HTMLElement
+
+  await step('Focusses first input', async () => {
+    await userEvent.click(button)
+    expect(firstInput).toHaveFocus()
+  })
+
+  await step('Does not focus first input', async () => {
+    await userEvent.click(button)
+    expect(secondInput).not.toHaveFocus()
+  })
+}
+
+export const FocusInvalid = () => ({
+  components: { VaForm, VaInput, VaButton },
+  template: `
+    <va-form 
+      class="flex space-x-2 mb-6 max-w-sm" 
+      ref="focusInvalidForm"
+    >
+      <va-input 
+        label="valid"
+      />
+      <va-input 
+        label="invalid"
+        error 
+      />
+    </va-form>
+    <va-button @click="$refs.focusInvalidForm.focusInvalidField()">
+      Focus invalid
+    </va-button>
+  `,
+})
+
+FocusInvalid.play = async ({ canvasElement, step }) => {
+  const canvas = within(canvasElement)
+  const firstInput = canvas.getByRole('textbox', { name: 'valid' }) as HTMLElement
+  const secondInput = canvas.getByRole('textbox', { name: 'invalid' }) as HTMLElement
+  const button = canvas.getByRole('button', { name: 'Focus invalid' }) as HTMLElement
+
+  await step('Focusses invalid input', async () => {
+    await userEvent.click(button)
+    expect(secondInput).toHaveFocus()
+  })
+
+  await step('Does not focus invalid input', async () => {
+    await userEvent.click(button)
+    expect(firstInput).not.toHaveFocus()
+  })
+}
+
+export const ValidateAndResetValidation = () => ({
+  components: { VaForm, VaInput, VaButton },
+  data: () => ({ input: 'do not reset' }),
+  template: `
+    <va-form 
+      class="flex space-x-2 mb-6 max-w-sm" 
+      ref="validateAndResetValidationForm"
+    >
+      <va-input
```

Comment:
I don't think we need this input here.

It basically checks "validate method doesn't validate inputs without rules". Which is weird thing to check and also has no particular relation to method.

In this story we only need to check that methods do what they meant to do.